include/rosdiscover-clang/BackwardSymbolizer/FloatSymbolizer.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+#include <clang/AST/Expr.h>
+#include <clang/AST/ExprCXX.h>
+#include <clang/AST/APValue.h>
+
+#include "../Builder/ValueBuilder.h"
+#include "../Value/Float.h"
+#include "../Helper/FindDefVisitor.h"
+
+namespace rosdiscover {
+
+class FloatSymbolizer {
+public:
+  FloatSymbolizer(
+  )
+  : valueBuilder() {}
+
+  std::unique_ptr<SymbolicFloat> symbolize(clang::Expr *expr) {
+
+    if (expr == nullptr) {
```

Comment:
It's weird that nullptr is even being passed in here. At the very least, I would write a big "ERROR" message to the stderr. (Otherwise you might be hiding bugs elsewhere in the analysis.)

Reply:
fixed